src/black/mode.py

```diff
@@ -298,3 +292,18 @@ def get_cache_key(self) -> str:
             features_and_magics,
         ]
         return ".".join(parts)
+
+    def __hash__(self) -> int:
```

Comment:
Why is this necessary? And could we instead make dataclass generate the hash method?